homeassistant/components/media_player/gpmdp.py

```diff
@@ -43,18 +44,41 @@ def __init__(self, name, address, create_connection):
         self._address = address
         self._name = name
         self._status = STATE_OFF
+        self._ws = None
+        self._title = None
+        self._artist = None
+        self._albumart = None
         self.update()
 
+    def get_ws(self):
+        """Check if the websocket is setup and connected."""
+        if self._ws is None:
+            try:
+                self._ws = self._connection(("ws://" + self._address + ":5672"),
+                                            timeout=1)
+            except (socket.timeout, ConnectionRefusedError, ConnectionResetError):
+                self._status = STATE_OFF
```

Comment:
This shouldn't be a side effect of retrieving the websocket but instead be done in `update`